sdk/servicebus/service-bus/test/perf/rhea-promise/send.ts

```diff
@@ -85,16 +86,24 @@ async function RunTest(
   await connection.close();
 }
 
-async function ExecuteSendsAsync(sender: any, messages: number): Promise<void> {
+async function ExecuteSendsAsync(sender: Sender, messages: number): Promise<void> {
+  sender.setMaxListeners(1000);
+  const onSuccessOrFail: Func<EventContext, void> = (context: EventContext) => {
+    canSend = true;
+    sender.removeListener(SenderEvents.rejected, onSuccessOrFail);
+    sender.removeListener(SenderEvents.accepted, onSuccessOrFail);
+  };
   while (++_messages <= messages) {
-    while (!sender.sendable()) {
+    while (!sender.sendable() || canSend === false) {
       await delay(0.01);
     }
     if (sender.sendable()) {
+      sender.once(SenderEvents.accepted, onSuccessOrFail);
+      sender.once(SenderEvents.rejected, onSuccessOrFail);
       await sender.send({ body: _payload });
+      canSend = false;
```

Comment:
The `sender.send()` call above does not return a promise, so no need to await on this.

Reply:
Do we run `tslint` on these perf tests?  If so, I believe this rule should have caught this error (assuming the rule is enabled):

https://palantir.github.io/tslint/rules/await-promise/

Reply:
The `await` here was really confusing me, since I assumed if `send()` returned a promise, it wouldn't resolve until the send was completely finished.  The fact that `send()` does **not** return a promise explains a lot.